runtime/io.c

```diff
@@ -672,21 +680,18 @@ CAMLprim value caml_ml_close_channel(value vchannel)
   /* For output channels, must have flushed before */
   struct channel * channel = Channel(vchannel);
 
-  /* Ensure that every read or write on the channel will cause an
-     immediate caml_flush_partial or caml_refill, thus raising a Sys_error
-     exception */
-  channel->curr = channel->max = channel->end;
```

Comment:
This deletion changes the behavior of runtime warnings: closing a channel no longer disables the warning about unflushed contents.
For instance, this program only emits warnings once this PR applied.
```ocaml
external close : out_channel -> unit = "caml_ml_close_channel"
let f s =
  let o = Out_channel.open_bin ("/tmp/"^s) in
  Out_channel.set_buffered o true;
  Out_channel.output_string o "test";
  close o
let () = f "a"
let () = for i = 0 to 1000 do let a = Array.make 1_000 () in () done
```

Reply:
I'm not aware of these runtime warnings, what form do they take?

Reply:
@Octachron I can't find a difference when running your example with and without this PR. Could you share your compilation commands?

Reply:
It is simpler to enable the warnings in the code itself, I have fixed the test program along this line. With trunk, there is no warning emitted, whereas I see the following warning with this PR:
```
[ocaml] (use Sys.enable_runtime_warnings to control these warnings)
[ocaml] (moreover, it has unflushed data)
```

Reply:
Thanks for the report, I hadn't seen this. I updated this function, it now resets `channel->curr` to `channel->buff` so that this false warning is not triggered. It seems clearer to me than re-introducing the deleted statement.

Reply:
Resetting `channel->curr` has the disadvantage of making it possible to write on the buffer of the closed channel without raising an error for a time.
Taking in account that
- Putch only flushes the channel when `channel->current`≥`channel->end`.
- Getch only refillls the channel when `channel->current` ≥ `channel->max` 
I am not sure if there is a simpler option than the existing
```C
channel->curr = channel->max = channel->end
```
to forbids all writes or reads on the disconnected buffer.

Reply:
Ah yes, you're right, I misunderstood that comment. I went back to the existing
```
channel->curr = channel->max = channel->end